File: 925 Long Pressed Name.cpp
class Solution {
public:
    bool isLongPressedName(string name, string typed) {
        int readName = 0;
        for(int i = 0;i < typed.size() && readName != name.size();++i){
            if(typed[i] == name[readName]){
                ++readName;
            }
        }
        return readName == name.size();
    }
};

File: 925 Long Pressed Name.go
func isLongPressedName(name string, typed string) bool {
    r := 0
    for i := 0;i < len(typed) && r != len(name);i++{
        if typed[i] == name[r]{
            r++
        }
    }
    return r == len(name)
}